In [2]:
import pandas as pd
import numpy as np

In [3]:
dataset=pd.read_csv("./cinemaTicket_Ref.csv",delimiter=",")
dataset.head()

film_code  cinema_code  total_sales  tickets_sold  tickets_out  show_time  \
0       1492          304      3900000            26            0          4   
1       1492          352      3360000            42            0          5   
2       1492          489      2560000            32            0          4   
3       1492          429      1200000            12            0          1   
4       1492          524      1200000            15            0          3   

   occu_perc  ticket_price  ticket_use    capacity        date  month  \
0       4.26      150000.0          26  610.328638  2018-05-05      5   
1       8.08       80000.0          42  519.801980  2018-05-05      5   
2      20.00       80000.0          32  160.000000  2018-05-05      5   
3      11.01      100000.0          12  108.991826  2018-05-05      5   
4      16.67       80000.0          15   89.982004  2018-05-05      5   

   quarter  day  
0        2    5  
1        2    5  
2        2    5  
3        2    5  
4        2    5

## First step is ayeball the data 
#### Let see what insight that we can find and of course usefull

In [4]:
dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 142524 entries, 0 to 142523
Data columns (total 14 columns):
 #   Column        Non-Null Count   Dtype  
---  ------        --------------   -----  
 0   film_code     142524 non-null  int64  
 1   cinema_code   142524 non-null  int64  
 2   total_sales   142524 non-null  int64  
 3   tickets_sold  142524 non-null  int64  
 4   tickets_out   142524 non-null  int64  
 5   show_time     142524 non-null  int64  
 6   occu_perc     142399 non-null  float64
 7   ticket_price  142524 non-null  float64
 8   ticket_use    142524 non-null  int64  
 9   capacity      142399 non-null  float64
 10  date          142524 non-null  object 
 11  month         142524 non-null  int64  
 12  quarter       142524 non-null  int64  
 13  day           142524 non-null  int64  
dtypes: float64(3), int64(10), object(1)
memory usage: 15.2+ MB


In [5]:
dataset.columns

Index(['film_code', 'cinema_code', 'total_sales', 'tickets_sold',
       'tickets_out', 'show_time', 'occu_perc', 'ticket_price', 'ticket_use',
       'capacity', 'date', 'month', 'quarter', 'day'],
      dtype='object')

In [6]:
subset_1 = dataset[['film_code','cinema_code','total_sales','ticket_price','date', 'month','day']]
subset_1

film_code  cinema_code  total_sales  ticket_price        date  month  \
0            1492          304      3900000      150000.0  2018-05-05      5   
1            1492          352      3360000       80000.0  2018-05-05      5   
2            1492          489      2560000       80000.0  2018-05-05      5   
3            1492          429      1200000      100000.0  2018-05-05      5   
4            1492          524      1200000       80000.0  2018-05-05      5   
...           ...          ...          ...           ...         ...    ...   
142519       1569          495      1320000       60000.0  2018-11-04     11   
142520       1569          474      1200000       80000.0  2018-11-04     11   
142521       1569          524      1060000      132500.0  2018-11-04     11   
142522       1569          529       600000      120000.0  2018-11-04     11   
142523       1569          486       250000       50000.0  2018-11-04     11   

        day  
0         5  
1         5  
2         5  
3         5  
4         5  
...     ...  
142519    4  
142520    4  
142521    4  
142522    4  
142523    4  

[142524 rows x 7 columns]

## So I looking for data that can increase sales 

In [7]:
subset_1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 142524 entries, 0 to 142523
Data columns (total 7 columns):
 #   Column        Non-Null Count   Dtype  
---  ------        --------------   -----  
 0   film_code     142524 non-null  int64  
 1   cinema_code   142524 non-null  int64  
 2   total_sales   142524 non-null  int64  
 3   ticket_price  142524 non-null  float64
 4   date          142524 non-null  object 
 5   month         142524 non-null  int64  
 6   day           142524 non-null  int64  
dtypes: float64(1), int64(5), object(1)
memory usage: 7.6+ MB


In [8]:
## The data is have good quality I'm not find missing values 
## I just need to change the Dtype of date

subset_1["date"]=pd.to_datetime(subset_1["date"])
subset_1.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 142524 entries, 0 to 142523
Data columns (total 7 columns):
 #   Column        Non-Null Count   Dtype         
---  ------        --------------   -----         
 0   film_code     142524 non-null  int64         
 1   cinema_code   142524 non-null  int64         
 2   total_sales   142524 non-null  int64         
 3   ticket_price  142524 non-null  float64       
 4   date          142524 non-null  datetime64[ns]
 5   month         142524 non-null  int64         
 6   day           142524 non-null  int64         
dtypes: datetime64[ns](1), float64(1), int64(5)
memory usage: 7.6 MB


C:\Users\ganoa\AppData\Local\Temp\ipykernel_1180\2956486709.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  subset_1["date"]=pd.to_datetime(subset_1["date"])


In [9]:
## Import the dataset so I can apply to BI tools PoweBI or Tableau
subset_1.to_csv("Cinema_Sales_Analysis.csv",index=False)

In [10]:
## Now I want to doing an simple analysis 
## Let see what we can get

In [11]:
cinema = subset_1

In [12]:
cinema.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 142524 entries, 0 to 142523
Data columns (total 7 columns):
 #   Column        Non-Null Count   Dtype         
---  ------        --------------   -----         
 0   film_code     142524 non-null  int64         
 1   cinema_code   142524 non-null  int64         
 2   total_sales   142524 non-null  int64         
 3   ticket_price  142524 non-null  float64       
 4   date          142524 non-null  datetime64[ns]
 5   month         142524 non-null  int64         
 6   day           142524 non-null  int64         
dtypes: datetime64[ns](1), float64(1), int64(5)
memory usage: 7.6 MB


In [13]:
cinema.columns.values

array(['film_code', 'cinema_code', 'total_sales', 'ticket_price', 'date',
       'month', 'day'], dtype=object)

In [14]:
sales_by_cinema = cinema [['cinema_code','total_sales']]
sales_by_film = cinema [['film_code','total_sales']] 
sales_by_ticket_price = cinema [['ticket_price','total_sales']]
sales_by_month = cinema [['month','total_sales']]
sales_by_day = cinema [['day','total_sales']]

In [15]:
sales_by_cinema.head()

cinema_code  total_sales
0          304      3900000
1          352      3360000
2          489      2560000
3          429      1200000
4          524      1200000

## Working with sales_by_cinema

In [16]:
sales_by_cinema["total_sales"].mean()

12347275.407531364

In [17]:
sales_by_cinema["total_sales"].median()

3720000.0

In [18]:
sales_by_cinema.max()

cinema_code           637
total_sales    1262819994
dtype: int64

In [19]:
sales_by_cinema.min()

cinema_code       32
total_sales    20000
dtype: int64

In [20]:
test=sales_by_cinema.value_counts()

In [21]:
(sales_by_cinema["cinema_code"].value_counts())

524    2474
489    2407
448    2345
163    2241
518    2193
       ... 
520      12
565      10
637       5
566       2
424       1
Name: cinema_code, Length: 246, dtype: int64

### So I want to sum "Total_sales" grouping by the each cinema
#### I'm Stuck LOL so we need to use simple data so maybe I can figure It out

In [22]:
raw_data = {'RegionCode' : ['10001', '10001', '10001', '10001', '10001', '10001', '10002', '10002', '10002', '10002', '10002', '10002'],
            'Stratum' : ['1', '1','2','2','3', '3', '1', '1', '2', '2', '3', '3'],
            'LaStratum' : ['1021', '1021', '1022', '1022', '1023', '1023', '2021', '2021', '2022', '2022', '2023', '2023'],
            'StratumPop' : [125, 125, 50, 50, 100, 100, 250, 250, 200, 200, 300, 300],
            'Q_response' : [2, 1, 4, 1, 2, 2, 3, 4, 3, 2, 1, 4]}


In [23]:
raw_data

{'RegionCode': ['10001',
  '10001',
  '10001',
  '10001',
  '10001',
  '10001',
  '10002',
  '10002',
  '10002',
  '10002',
  '10002',
  '10002'],
 'Stratum': ['1', '1', '2', '2', '3', '3', '1', '1', '2', '2', '3', '3'],
 'LaStratum': ['1021',
  '1021',
  '1022',
  '1022',
  '1023',
  '1023',
  '2021',
  '2021',
  '2022',
  '2022',
  '2023',
  '2023'],
 'StratumPop': [125, 125, 50, 50, 100, 100, 250, 250, 200, 200, 300, 300],
 'Q_response': [2, 1, 4, 1, 2, 2, 3, 4, 3, 2, 1, 4]}

In [24]:
type(raw_data)

dict

In [25]:
Data=pd.DataFrame({'RegionCode' : ['10001', '10001', '10001', '10001', '10001', '10001', '10002', '10002', '10002', '10002', '10002', '10002'],
            'Stratum' : ['1', '1','2','2','3', '3', '1', '1', '2', '2', '3', '3'],
            'LaStratum' : ['1021', '1021', '1022', '1022', '1023', '1023', '2021', '2021', '2022', '2022', '2023', '2023'],
            'StratumPop' : [125, 125, 50, 50, 100, 100, 250, 250, 200, 200, 300, 300],
            'Q_response' : [2, 1, 4, 1, 2, 2, 3, 4, 3, 2, 1, 4]})

In [26]:
Data

RegionCode Stratum LaStratum  StratumPop  Q_response
0       10001       1      1021         125           2
1       10001       1      1021         125           1
2       10001       2      1022          50           4
3       10001       2      1022          50           1
4       10001       3      1023         100           2
5       10001       3      1023         100           2
6       10002       1      2021         250           3
7       10002       1      2021         250           4
8       10002       2      2022         200           3
9       10002       2      2022         200           2
10      10002       3      2023         300           1
11      10002       3      2023         300           4

In [27]:
xxx = Data.groupby('RegionCode')['StratumPop'].sum()

In [28]:
xxx
### Awasome I find the syntax COOL!!!

RegionCode
10001     550
10002    1500
Name: StratumPop, dtype: int64

In [29]:
### So now I want create another dataframe for sum of total sales by each cinema 
### So I can know what cinema is the most profitable

In [30]:
sales_by_cinema.head()

cinema_code  total_sales
0          304      3900000
1          352      3360000
2          489      2560000
3          429      1200000
4          524      1200000

In [31]:
cinema_by_revenue = sales_by_cinema.groupby ("cinema_code")["total_sales"].sum()
cinema_by_revenue

cinema_code
32      9273549998
33      3947850000
34     26891660000
35     12952150000
39     13126359999
          ...     
561     1120946000
562      204660000
565        9680000
566        4000000
637     1220000000
Name: total_sales, Length: 246, dtype: int64

In [32]:
type(cinema_by_revenue)

pandas.core.series.Series

In [33]:
total_cinema_revenue =pd.DataFrame(cinema_by_revenue)

In [34]:
cinema_by_revenue.sort_values()

cinema_code
566         4000000
543         9510000
565         9680000
424        10050000
542        18780000
           ...     
489     58831047000
524     59306304992
163     62852889999
304    101701033000
448    214230699820
Name: total_sales, Length: 246, dtype: int64

In [35]:
sales_by_cinema.min()

cinema_code       32
total_sales    20000
dtype: int64

## Information I get from sales_by_cinema
#### mean = 12347275
#### median = 3720000
#### cinema(max) singel day = 637 with revenue 1262819994
#### cinema(min) singel day = 32 with revenue 20000
#### total revenue (max) = cinema 448 with total revenue 214230699820
#### total revenue (min) = cinema 566 with total revenue 4000000

In [37]:
total_cinema_revenue

total_sales
cinema_code             
32            9273549998
33            3947850000
34           26891660000
35           12952150000
39           13126359999
...                  ...
561           1120946000
562            204660000
565              9680000
566              4000000
637           1220000000

[246 rows x 1 columns]

### Good I get what I want so save the file 
#### Let see the visualization in BI tools

In [39]:
sales_by_cinema.to_csv("Sales_By_Cinema.csv",index=False)
total_cinema_revenue.to_csv("Total_Cinema_Revenue.csv",index=False)

## CheckPoint

# Working with sales_by_film

In [ ]:
# Working with
# sales_by_cinema = cinema [['cinema_code','total_sales']]
# sales_by_film = cinema [['film_code','total_sales']] 
# sales_by_ticket_price = cinema [['ticket_price','total_sales']]
# sales_by_month = cinema [['month','total_sales']]
# sales_by_day = cinema [['day','total_sales']]

In [40]:
sales_by_film
## Fint total sales by each film

film_code  total_sales
0            1492      3900000
1            1492      3360000
2            1492      2560000
3            1492      1200000
4            1492      1200000
...           ...          ...
142519       1569      1320000
142520       1569      1200000
142521       1569      1060000
142522       1569       600000
142523       1569       250000

[142524 rows x 2 columns]

In [41]:
sales_by_film["total_sales"].mean()

12347275.407531364

In [42]:
sales_by_film.max()

film_code            1589
total_sales    1262819994
dtype: int64

In [43]:
sales_by_film.min()

film_code       1471
total_sales    20000
dtype: int64

In [44]:
total_film_revenue=sales_by_film.groupby("film_code")["total_sales"].sum()
total_film_revenue=pd.DataFrame(total_film_revenue)

In [45]:
total_film_revenue.max()
total_film_revenue.min()

total_sales    16490000
dtype: int64

## Information I get from sales_by_film
#### mean = 12347275
#### median = 3720000
#### film (max) singel day = 1589 with revenue 1262819994
#### film(min) singel day = 1471 with revenue 20000
#### total revenue (max) =  with total revenue 372740376637
#### total revenue (min) =  with total revenue 16490000

# Check point

In [ ]:
sales_by_film.to_csv("Sales By Film.csv",index=False)
total_film_revenue.to_csv("Sales By Revenue.csv",index=False)

In [46]:
sales_by_day.head()

day  total_sales
0    5      3900000
1    5      3360000
2    5      2560000
3    5      1200000
4    5      1200000

In [47]:
sales_by_day["total_sales"].mean()

12347275.407531364

In [ ]:
sales_by_day.max()

day                    31
total_sales    1262819994
dtype: int64

In [48]:
sales_by_day.min()

day                1
total_sales    20000
dtype: int64

In [ ]:
total_weekday_revenue=sales_by_day.groupby("day")["total_sales"].sum()
total_weekday_revenue.sort_values()

day
8     40575913460
15    42260581942
9     42360278956
21    43076481442
7     46160391466
30    48573177964
20    49943669926
22    52222330440
5     52682814402
6     52800437426
23    52882677951
29    53194324445
31    53570589464
18    53941971421
16    54336397433
14    54744399920
13    56372876903
11    58345596949
4     60111639961
25    60728857428
17    61848197937
12    62185741937
19    62528090455
28    63318501938
3     63505859214
2     64359762203
10    65960359945
1     68195297468
27    72595696437
26    73197241419
24    73202921931
Name: total_sales, dtype: int64

## So its looking good its by day of the month but I need something like 
## what day on the weekend is most sales
## OK LETS CRACK IT

In [49]:
## I find this code let see if this work
def date_to_weekday(date_value):
    return date_value.weekday()

In [50]:
## So we need to suset the data column 
## Make sure the Dtype is timestam
cinema.head()

film_code  cinema_code  total_sales  ticket_price       date  month  day
0       1492          304      3900000      150000.0 2018-05-05      5    5
1       1492          352      3360000       80000.0 2018-05-05      5    5
2       1492          489      2560000       80000.0 2018-05-05      5    5
3       1492          429      1200000      100000.0 2018-05-05      5    5
4       1492          524      1200000       80000.0 2018-05-05      5    5

In [52]:
sales_by_weekday=cinema[["date","total_sales"]]
sales_by_weekday["day of the week"]=sales_by_weekday["date"].apply(date_to_weekday)


C:\Users\ganoa\AppData\Local\Temp\ipykernel_1180\1030861419.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  sales_by_weekday["day of the week"]=sales_by_weekday["date"].apply(date_to_weekday)


In [53]:
sales_by_weekday.tail()
## OK Awesome Cool 
## lets check it

date  total_sales  day of the week
142519 2018-11-04      1320000                6
142520 2018-11-04      1200000                6
142521 2018-11-04      1060000                6
142522 2018-11-04       600000                6
142523 2018-11-04       250000                6

In [54]:
sales_by_weekday["day of the week"].unique()

array([5, 4, 3, 2, 6, 1, 0], dtype=int64)

In [55]:
### Good now just need to drop the date column
sales_by_weekday.drop(columns=["date"],inplace=True)

C:\Users\ganoa\AppData\Local\Temp\ipykernel_1180\2626596763.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  sales_by_weekday.drop(columns=["date"],inplace=True)


In [56]:
total_revenue_by_day=sales_by_weekday.groupby("day of the week")["total_sales"].sum()

In [57]:
total_revenue_by_day.sort_values()

day of the week
0    137003523358
6    158196914862
5    194059936231
2    225169454951
1    317197735361
3    335440301328
4    392715214092
Name: total_sales, dtype: int64

In [58]:
type(total_revenue_by_day)

pandas.core.series.Series

## Check Point

In [59]:
sales_by_month

month  total_sales
0           5      3900000
1           5      3360000
2           5      2560000
3           5      1200000
4           5      1200000
...       ...          ...
142519     11      1320000
142520     11      1200000
142521     11      1060000
142522     11       600000
142523     11       250000

[142524 rows x 2 columns]

In [64]:
sales_by_month["month"].unique()

array([ 5,  8,  7,  6,  9, 10, 11,  3,  4,  2], dtype=int64)

In [65]:
## Looking for which month with most sales
sales_by_month.groupby("month")["total_sales"].sum()

month
2         32210000
3     193433625732
4     348614664007
5     239595373317
6     104001473893
7     264855462688
8     258296800764
9      96569594964
10    231675849820
11     22708024998
Name: total_sales, dtype: int64

In [66]:
total_revenue_by_month = sales_by_month.groupby("month")["total_sales"].sum()
total_revenue_by_month.sort_values()

month
2         32210000
11     22708024998
9      96569594964
6     104001473893
3     193433625732
10    231675849820
5     239595373317
8     258296800764
7     264855462688
4     348614664007
Name: total_sales, dtype: int64

## Good month 4 is most revenue, CheckPoint

In [67]:
sales_by_ticket_price

ticket_price  total_sales
0           150000.0      3900000
1            80000.0      3360000
2            80000.0      2560000
3           100000.0      1200000
4            80000.0      1200000
...              ...          ...
142519       60000.0      1320000
142520       80000.0      1200000
142521      132500.0      1060000
142522      120000.0       600000
142523       50000.0       250000

[142524 rows x 2 columns]

In [68]:
sales_by_ticket_price["ticket_price"].mean()

81234.59988649565

In [70]:
sales_by_ticket_price["ticket_price"].max()

700000.0

In [71]:
sales_by_ticket_price["ticket_price"].min()

483.8709677419355

In [72]:
sales_by_ticket_price.groupby("ticket_price")["total_sales"].sum()

ticket_price
483.870968          90000
1600.000000        160000
2000.000000         80000
2061.855670        800000
2285.714286         80000
                   ...   
204444.444444     1840000
280000.000000     1120000
470000.000000    10340000
500000.000000    12500000
700000.000000     4200000
Name: total_sales, Length: 23235, dtype: int64

In [73]:
total_revenuy_by_ticket_price = sales_by_ticket_price.groupby("ticket_price")["total_sales"].sum()

In [74]:
total_revenuy_by_ticket_price.sort_values()

ticket_price
2500.000000             40000
4000.000000             56000
2000.000000             80000
2285.714286             80000
2424.242424             80000
                     ...     
60000.000000      72934140000
120000.000000     77194800000
100000.000000     97860800000
150000.000000    131666100000
80000.000000     177008240000
Name: total_sales, Length: 23235, dtype: int64

## Awesome I find some insight that can help for decisions
#### Like what range price is generate most revenue
#### What month & day that good for release some movie 
#### What cinema that with most revenue 
#### And what film is generate most revenue

## So disclaimer this simple analisis is purpose for me or you to learn how to clean and preprocesing data and some analisis
## That can help data sains or management to make a decision 